http://bugs.gentoo.org/157395
http://gcc.gnu.org/PR30185

2006-12-15  Jakub Jelinek  <jakub@redhat.com>

	PR target/30185
	* config/rs6000/rs6000.md (div<mode>3): Use correct mode in
	force_reg call.

--- branches/gcc-4_1-branch/gcc/config/rs6000/rs6000.md	2006/12/15 23:09:26	119948
+++ branches/gcc-4_1-branch/gcc/config/rs6000/rs6000.md	2006/12/15 23:11:00	119949
@@ -1883,7 +1883,7 @@
     ;
   else if (TARGET_POWERPC)
     {
-      operands[2] = force_reg (SImode, operands[2]);
+      operands[2] = force_reg (<MODE>mode, operands[2]);
       if (TARGET_POWER)
 	{
 	  emit_insn (gen_divsi3_mq (operands[0], operands[1], operands[2]));
